script/testing/oltpbench/run_oltpbench_v2.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python3
+
+import os
+import sys
+import argparse
+import traceback
+import json
+import itertools
+
+base_path = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+sys.path.insert(0, base_path)
+
+from oltpbench.test_case_oltp import TestCaseOLTP
+from oltpbench.test_oltpbench_v2 import TestOLTPBenchV2
+
+def parse_tracefile(aparser, args):
+    aparser.add_argument(
+        "tracefile", help="File containing a collection of test cases")
+    aparser.add_argument("--db-host", help="DB Hostname")
+    aparser.add_argument("--db-port", type=int, help="DB Port")
+    aparser.add_argument("--db-output-file", help="DB output log file")
+    aparser.add_argument("--build-type",
+                         default="debug",
+                         choices=["debug", "release", "relwithdebinfo"],
+                         help="Build type (default: %(default)s")
+    args = vars(aparser.parse_args(args))
+
+    args_tracefile = args.get("tracefile")
+
+    oltp_test_suite = []
+
+    if args_tracefile:
+        tracefile_path = os.path.join(os.getcwd(), args_tracefile)
+        if not os.path.exists(tracefile_path):
+            msg = "Unable to find OLTPBench trace file '{}'".format(
+                args_tracefile)
+            raise RuntimeError(msg)
+
+        with open(tracefile_path) as oltp_tracefile:
+            oltp_test_suite_json = json.load(oltp_tracefile)
+
+        for oltp_testcase in oltp_test_suite_json["testcases"]:
```

Comment:
agree. we can set it as `None` outside and load the json contents as dict to `oltp_test_suite_json`. After that we can check if it is `None` again before the for loop. 

Think about a case: what if a file exists but
- it is an empty file
- it is not a valid json file (well `json.load` can handle such exception)
- it is a valid json file but with empty object (like `{}` or `[]`)

Reply:
BTW, try to stop using `oltp_test_suite_json["testcases"]` way to access an element from a dict in Python. It is **unsafe** since the key `"testcases"` may not even exist. If that happens, `KeyError` will be thrown and there is nowhere we can handle it. Try to use `oltp_test_suite_json.get("testcases")` instead since the `.get` function of the Python dict can return `None` as default value. In this for loop case, we can do something like `oltp_test_suite_json.get("testcases", [])` to return an empty list as the default value.